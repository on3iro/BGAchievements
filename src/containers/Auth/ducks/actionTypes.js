export const LOGIN_SUBMIT = 'LOGIN_SUBMIT'
export const LOGIN_REQUEST = 'LOGIN_REQUEST'
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS'
export const LOGIN_ERROR = 'LOGIN_ERROR'
export const LOGOUT = 'LOGOUT'
export const REGISTER_SUBMIT = 'REGISTER_SUBMIT'
export const REGISTER_REQUEST = 'REGISTER_REQUEST'
export const REGISTER_SUCCESS = 'REGISTER_SUCCESS'
export const REGISTER_ERROR = 'REGISTER_ERROR'
export const UPDATE_MAX_ACCOUNTS = 'UPDATE_MAX_ACCOUNTS'
