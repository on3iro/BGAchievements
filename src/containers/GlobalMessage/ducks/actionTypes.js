export const GLOBAL_ERROR = 'GLOBAL_ERROR'
export const GLOBAL_MESSAGE = 'GLOBAL_MESSAGE'
export const CLEAR_MESSAGE = 'CLEAR_MESSAGE'
