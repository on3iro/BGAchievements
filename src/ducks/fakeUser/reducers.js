// @flow

import { loop, Cmd } from 'redux-loop'

import {
  INIT,
  USER_FETCH_FAIL,
  USER_FETCH_SUCCESS
} from './constants'

import type {
  Action,
  State
} from './types'

import {
  fetchUser,
  userFetchFailedAction,
  userFetchSuccessfulAction
} from './actions'

const fakeUserReducer = (state: State = {
  loading: false,
  user: null,
  error: null
}, action: Action): State => {
  switch (action.type) {
    case INIT:
      return loop(
        {...state, loading: true},
        Cmd.run(fetchUser, {
          successActionCreator: userFetchSuccessfulAction,
          failActionCreator: userFetchFailedAction,
          args: ['userA']
        })
      )

    case USER_FETCH_SUCCESS:
      return { ...state, user: action.payload, loading: false }

    case USER_FETCH_FAIL:
      return { ...state, error: action.payload, loading: false }

    default:
      return state
  }
}

export {
  fakeUserReducer
}
