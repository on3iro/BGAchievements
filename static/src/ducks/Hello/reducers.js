import { combineReducers, loop, Cmd } from 'redux-loop'

import * as actions from './actions'
import sideeffects from './sideeffects'

const HelloReducer = (state = [], action) => {
  switch (action.type) {
    case 'SAMPLE_DATA_FETCH_START': {
      return loop(state, Cmd.run(sideeffects.fetchSampleData, {
        successActionCreator: actions.fetchSampleDataSuccess,
        errorActionCreator: actions.fetchSampleDataFailure
      }))
    }

    case 'SAMPLE_DATA_FETCH_SUCCESS': {
      return action.payload
    }

    case 'SAMPLE_DATA_FETCH_FAILURE': {
      return []
    }

    default: {
      return state
    }
  }
}

const LoadingReducer = (state = false, action) => {
  switch (action.type) {
    case 'SAMPLE_DATA_FETCH_START': {
      return true
    }

    case 'SAMPLE_DATA_FETCH_SUCCESS':
    case 'SAMPLE_DATA_FETCH_FAILURE': {
      return false
    }

    default: {
      return state
    }
  }
}

export default combineReducers({
  sampleData: HelloReducer,
  loading: LoadingReducer
})
