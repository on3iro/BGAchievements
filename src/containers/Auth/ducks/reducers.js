import { combineReducers } from 'redux'

import * as types from './actionTypes'

export const user = (state = {
  isLoggedIn: false,
  hasRegistered: false,
  token: '',
  data: {},
  error: {}
}, action) => {
  switch (action.type) {
    case types.LOGIN_SUCCESS: {
      const data = action.payload

      return {
        ...state,
        isLoggedIn: true,
        token: data.token,
        data: data.user,
        error: ''
      }
    }

    case types.REGISTER_SUCCESS: {
      return {
        ...state,
        hasRegistered: true
      }
    }

    case types.REGISTER_ERROR:
    case types.LOGIN_ERROR: {
      const error = action.payload

      return {
        ...state,
        error: error
      }
    }

    case types.LOGOUT: {
      return {
        ...state,
        isLoggedIn: false,
        token: '',
        data: {}
      }
    }

    case types.UPDATE_MAX_ACCOUNTS: {
      return {
        ...state,
        data: {
          ...state.data,
          maxAccounts: action.payload
        }
      }
    }

    default: {
      return state
    }
  }
}

export const AuthRequestState = (state = {
  isLoading: false
}, action) => {
  switch (action.type) {
    case types.LOGIN_SUBMIT: {
      return { ...state, isLoading: true }
    }

    case types.REGISTER_SUBMIT: {
      return { ...state, isLoading: true }
    }

    case types.LOGIN_SUCCESS: {
      return { ...state, isLoading: false }
    }

    case types.REGISTER_SUCCESS: {
      return { ...state, isLoading: false }
    }

    case types.REGISTER_ERROR: {
      return { ...state, isLoading: false }
    }

    case types.LOGIN_ERROR: {
      return { ...state, isLoading: false }
    }

    default:
      return state
  }
}

const authReducer = combineReducers({
  user,
  AuthRequestState
})

export default authReducer
