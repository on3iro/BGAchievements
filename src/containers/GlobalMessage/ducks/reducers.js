import { combineReducers } from 'redux'

import * as types from './actionTypes'

export const GlobalError = (state = '', action) => {
  switch (action.type) {
    case types.GLOBAL_ERROR: {
      return action.payload
    }

    case types.CLEAR_MESSAGE: {
      return ''
    }

    default: {
      return state
    }
  }
}

export const GlobalMessage = (state = '', action) => {
  switch (action.type) {
    case types.GLOBAL_MESSAGE: {
      return action.payload
    }

    case types.CLEAR_MESSAGE: {
      return ''
    }

    default: {
      return state
    }
  }
}

const GlobalMessages = combineReducers({
  GlobalError,
  GlobalMessage
})

export default GlobalMessages
