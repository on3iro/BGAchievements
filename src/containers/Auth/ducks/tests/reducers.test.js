import * as reducers from '../reducers'
import * as types from '../actionTypes'

describe('authReducer', () => {
  describe('userReducer', () => {
    it('should return initial state', () => {
      expect(reducers.user(undefined, {})).toMatchSnapshot()
    })

    it('should handle LOGIN_SUCCESS', () => {
      expect(reducers.user(undefined, {
        type: types.LOGIN_SUCCESS,
        payload: {
          token: 'testtoken',
          user: 'testuser'
        }
      })).toMatchSnapshot()
    })

    it('should handle REGISTER_SUCCESS', () => {
      expect(reducers.user(undefined, {
        type: types.REGISTER_SUCCESS
      })).toMatchSnapshot()
    })

    it('should handle REGISTER_ERROR', () => {
      expect(reducers.user(undefined, {
        type: types.REGISTER_ERROR,
        payload: 'register-error'
      })).toMatchSnapshot()
    })

    it('should handle LOGIN_ERROR', () => {
      expect(reducers.user(undefined, {
        type: types.LOGIN_ERROR,
        payload: 'login-error'
      })).toMatchSnapshot()
    })

    it('should handle LOGOUT', () => {
      expect(reducers.user({
        user: {
          isLoggedIn: true,
          token: 'testoken'
        }
      }, {
        type: types.LOGOUT
      })).toMatchSnapshot()
    })
  })

  describe('AuthRequestStateReducer', () => {
    it('should return initial state', () => {
      expect(reducers.AuthRequestState(undefined, {})).toMatchSnapshot()
    })

    it('should handle LOGIN_SUBMIT', () => {
      expect(reducers.AuthRequestState(undefined, {
        type: types.LOGIN_SUBMIT
      })).toMatchSnapshot()
    })

    it('should handle REGISTER_SUBMIT', () => {
      expect(reducers.AuthRequestState(undefined, {
        type: types.REGISTER_SUBMIT
      })).toMatchSnapshot()
    })

    it('should handle LOGIN_SUCCESS', () => {
      expect(reducers.AuthRequestState(undefined, {
        type: types.LOGIN_SUCCESS
      })).toMatchSnapshot()
    })

    it('should handle REGISTER_SUCCESS', () => {
      expect(reducers.AuthRequestState(undefined, {
        type: types.REGISTER_SUCCESS
      })).toMatchSnapshot()
    })

    it('should handle LOGIN_ERROR', () => {
      expect(reducers.AuthRequestState(undefined, {
        type: types.LOGIN_ERROR
      })).toMatchSnapshot()
    })

    it('should handle REGISTER_ERROR', () => {
      expect(reducers.AuthRequestState(undefined, {
        type: types.REGISTER_ERROR
      })).toMatchSnapshot()
    })
  })
})
