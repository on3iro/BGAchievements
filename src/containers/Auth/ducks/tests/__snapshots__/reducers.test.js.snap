exports[`authReducer AuthRequestStateReducer should handle LOGIN_ERROR 1`] = `
Object {
  "isLoading": false,
}
`

exports[`authReducer AuthRequestStateReducer should handle LOGIN_SUBMIT 1`] = `
Object {
  "isLoading": true,
}
`

exports[`authReducer AuthRequestStateReducer should handle LOGIN_SUCCESS 1`] = `
Object {
  "isLoading": false,
}
`

exports[`authReducer AuthRequestStateReducer should handle REGISTER_ERROR 1`] = `
Object {
  "isLoading": false,
}
`

exports[`authReducer AuthRequestStateReducer should handle REGISTER_SUBMIT 1`] = `
Object {
  "isLoading": true,
}
`

exports[`authReducer AuthRequestStateReducer should handle REGISTER_SUCCESS 1`] = `
Object {
  "isLoading": false,
}
`

exports[`authReducer AuthRequestStateReducer should return initial state 1`] = `
Object {
  "isLoading": false,
}
`

exports[`authReducer userReducer should handle LOGIN_ERROR 1`] = `
Object {
  "data": Object {},
  "error": "login-error",
  "hasRegistered": false,
  "isLoggedIn": false,
  "token": "",
}
`

exports[`authReducer userReducer should handle LOGIN_SUCCESS 1`] = `
Object {
  "data": "testuser",
  "error": "",
  "hasRegistered": false,
  "isLoggedIn": true,
  "token": "testtoken",
}
`

exports[`authReducer userReducer should handle LOGOUT 1`] = `
Object {
  "data": Object {},
  "isLoggedIn": false,
  "token": "",
  "user": Object {
    "isLoggedIn": true,
    "token": "testoken",
  },
}
`

exports[`authReducer userReducer should handle REGISTER_ERROR 1`] = `
Object {
  "data": Object {},
  "error": "register-error",
  "hasRegistered": false,
  "isLoggedIn": false,
  "token": "",
}
`

exports[`authReducer userReducer should handle REGISTER_SUCCESS 1`] = `
Object {
  "data": Object {},
  "error": Object {},
  "hasRegistered": true,
  "isLoggedIn": false,
  "token": "",
}
`

exports[`authReducer userReducer should return initial state 1`] = `
Object {
  "data": Object {},
  "error": Object {},
  "hasRegistered": false,
  "isLoggedIn": false,
  "token": "",
}
`
